plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/MockThriftHttpMetastoreServer.java

```diff
@@ -0,0 +1,1763 @@
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.eclipse.jetty.util.thread.ExecutorThreadPool;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadPoolExecutor;
+
+public class MockThriftHttpMetastoreServer
+{
+    private final Server jettyServer;
+    private final MockThriftHttpServelet thriftHttpServlet;
+    private int port;
+    public static final String httpPath = "/mock-end-point";
+
+    public MockThriftHttpMetastoreServer(InMemoryThriftMetastore delegate, TestHttpRequestInterceptor requestInterceptor)
+            throws Exception
+    {
+        ExecutorService executorService = Executors.newFixedThreadPool(10);
+        ExecutorThreadPool threadPool = new ExecutorThreadPool((ThreadPoolExecutor) executorService);
+
+        jettyServer = new Server(threadPool);
+        jettyServer.setStopAtShutdown(true);
+
+        ServerConnector connector = new ServerConnector(jettyServer, new HttpConnectionFactory());
+        port = getRandomAvailablePort();
+        connector.setPort(port);
+        connector.setReuseAddress(true);
+        jettyServer.addConnector(connector);
+
+        MockThriftHiveMetastoreHandler mockThriftHandler = new MockThriftHiveMetastoreHandler(delegate);
+        TProcessor processor = new ThriftHiveMetastore.Processor<>(mockThriftHandler);
+        thriftHttpServlet = new MockThriftHttpServelet(processor, new TBinaryProtocol.Factory(), Optional.of(requestInterceptor));
+        ServletContextHandler context = new ServletContextHandler(
+                ServletContextHandler.NO_SECURITY | ServletContextHandler.NO_SESSIONS);
+        jettyServer.setHandler(context);
+
+        context.addServlet(new ServletHolder(thriftHttpServlet), httpPath);
+        start();
+    }
+
+    public void start()
+            throws Exception
+    {
+        jettyServer.start();
+    }
+
+    public void stop()
+            throws Exception
+    {
+        jettyServer.stop();
+    }
+
+    @ResourcePresence
+    public boolean isRunning()
+    {
+        return jettyServer.isRunning();
+    }
+
+    private int getRandomAvailablePort()
+            throws IOException
+    {
+        int freePort = -1;
+        try (ServerSocket temp = new ServerSocket(0)) {
+            freePort = temp.getLocalPort();
+        }
+        return freePort;
+    }
+
+    public int getPort()
+    {
+        return port;
+    }
+
+    private static class MockThriftHttpServelet
+            extends TServlet
+    {
+        private final Optional<TestHttpRequestInterceptor> requestInterceptor;
```

Comment:
Why optional? This class is only used in one place and the interceptor is provided there.

Reply:
Makes sense.